docs: Add Incremental Build Vitepress docs (#1404)
SAP/ui5-tooling: internal/documentation/docs/updates/migrate-v5.md

- **Quick iterations**: Changes to individual files typically rebuild very quickly
- **Cross-session**: Caches are used between server restarts and build runs

### For `ui5 build`

When `ui5 build` is executed, build caches are automatically serialized and reused when available.

::: tip Tip for Single Builds (e.g. CI/CD)
If you plan to execute a build only once (e.g. during a CI run), consider using `--cache "Off"` (see [Build Cache Control](../pages/Builder.md#build-cache-control)) to skip cache serialization.
:::

### For `ui5 serve`

During a server session (started by `ui5 serve`), source changes are automatically being monitered. Then, if a request was made, the server detects this, tries to use caches and only rebuilds when none are available. For more information see [Watch Mode Behavior](../pages/Server.md#watch-mode-behavior).

::: info Review custom middleware

With this feature, **build tasks are executed in server sessions**. For this reason, some custom middleware might be obsolete or cause problems. Projects utilizing such might need to adapt their configuration.

> E.g. Middleware for browser live reloads is obsolete and can be removed. Also, middleware for Typescript transpilation is not required anymore (as long as the respective task is still set up).

Reviewer: - The most prominent difference is that the server now actually runs a build. So any middleware that performed work which also a task did, becomes obsolete. This is not limited to TypeScript. I think this fact should also be mentioned in the beginning (`Build Cache`). The simplification and making most (custom) middleware obsolete is one of the main reasons for this feature. We should highlight that.

- Live reload will be implemented via #1407, and that PR already includes a hint for that.

- It should be mentioned that middleware for non-build purposes such as proxies are not affected by this, and still a valid use case for a custom middleware

Author: Done

Projects using older **Specification Versions** are expected to be **fully compatible with UI5 CLI v5**.

## Build Cache

> UI5 CLI v5 introduces **builds with caching** for both commands `ui5 build` and `ui5 serve`. This fundamental architectural change significantly improves build performance by reusing cached results from previous builds and simplifies development with the Server by making most custom middleware obsolete.

Reviewer 2: ```suggestion
UI5 CLI v5 introduces **builds with caching** for both the `ui5 build` and `ui5 serve` commands. This fundamental architectural change significantly improves build performance by reusing cached results from previous builds. It also simplifies development with the server by making most custom middleware obsolete.
```

### What Changed

**Previous Behavior (v4 and earlier):**

> - Every source change resulted in an entire rebuild of all projects

Reviewer 2: ```suggestion
- Every source change resulted in a full rebuild of all projects
```

- Every build re-executed all tasks
- No caching between builds or server sessions

> - Custom middleware required for special actions

Reviewer 2: ```suggestion
- Custom middleware was required for special actions
```

**New Behavior (v5):**
- Only relevant projects are rebuilt
- Only modified resources and affected tasks are re-processed

> - Custom middleware obsolete by replacing with build tasks

Reviewer 2: ```suggestion
- Some custom middleware has been replaced with build tasks and is now obsolete
```

Author: This actually should state that most custom middleware is obsolete now and can be replaced with corresponding build tasks. Can we rephrase it to "can be replaced with.." instead of "has been replaced" ?

> - Custom middleware obsolete by replacing with build tasks

Reviewer 2: ```suggestion
- Some custom middleware is now obsolete as it can be replaced with build tasks
```

### Impact on Your Workflow

> - **First build**: Probably slightly slower, as the cache is populated with all build outputs

Reviewer 2: ```suggestion
- **First build**: Probably slightly slower as the cache is populated with all build outputs
```

- **Subsequent builds**: Significantly faster — only modified resources and affected tasks are reprocessed
- **Quick iterations**: Changes to individual files typically rebuild very quickly
- **Cross-session**: Caches are used between server restarts and build runs

### For `ui5 build`

When `ui5 build` is executed, build caches are automatically serialized and reused when available.

::: tip Tip for Single Builds (e.g. CI/CD)

> If you plan to execute a build only once (e.g. during a CI run), consider using `--cache "Off"` (see [Build Cache Control](../pages/Builder.md#build-cache-control)) to skip cache serialization.

Reviewer 2: ```suggestion
If you plan to execute a build only once (for example during a CI run), consider using `--cache "Off"` (see [Build Cache Control](../pages/Builder.md#build-cache-control)) to skip cache serialization.
```

:::

### For `ui5 serve`

The UI5 Server now performs a build of the project. When started with `ui5 serve`, a similar build to `ui5 build` is executed containing standard and custom tasks (see [exceptions](../pages/Server.md#standard-tasks)).

> During a server session, source changes are automatically being monitered. Then, if a request was made, the server detects this, tries to use caches and only rebuilds when none are available. For more information see [Watch Mode Behavior](../pages/Server.md#watch-mode-behavior).

Reviewer 2: ```suggestion
During a server session, source changes are automatically monitored. When a request is made, the server detects this, tries to use caches, and only rebuilds when none are available. For more information, see [Watch Mode Behavior](../pages/Server.md#watch-mode-behavior).
```

::: tip Review custom middleware

> Due to **build tasks now being executed in server sessions**, custom middleware becomes obsolete if a corresponding task can perform the same actions (e.g. Typescript transpilation). However, custom middleware for non-build purposes such as proxies are not affected by this, and are still a valid use case.

Reviewer 2: ```suggestion
Due to **build tasks now being executed in server sessions**, custom middleware becomes obsolete if a corresponding task can perform the same actions (for example, Typescript transpilation). However, custom middleware for non-build purposes, such as proxies, are not affected by this and are still a valid use case.
```

The following middleware has been removed from the [standard middlewares list](../pages/Server.md#standard-middleware):

> * `serveThemes` — Theme compilation (LESS to CSS) is now handled by the `buildThemes` build task. Due to server sessions now also performing builds, this task gets executed during a server start, rather than on demand during runtime. The resulting CSS files are served via the `serveResources` middleware. This change improves performance through build-time compilation and caching while maintaining the same functionality.

Reviewer 2: ```suggestion
* `serveThemes` — The `buildThemes` build task now handles theme compilation (LESS to CSS). Because server sessions now also perform builds, this task runs during a server start instead of on demand during runtime. The resulting CSS files are served by the `serveResources` middleware. This change improves performance through build-time compilation and caching while maintaining the same functionality.
```
